### PR TITLE
Testing light

### LINKED ISSUE #1620: Add buyable light source to menu
**Is your feature request related to a problem? Please describe.**
The night is dark and full of terrors.

**Describe the solution you'd like**
Add a buyable light source.

Possible lights: 
```
Land_PortableLight_double_F
Land_LampShabby_F
```

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
Adds a purchasable light source to the vehicle box. 
Allowing for a brighter HQ during the night.
Allows for seeing the map better at night during early game missions.
If someone wants to delete the light the just garage it and gets the refund.

    
Added a vehiclelightsource variable to the rebel templates, allowing for the change to a different 
light source object if players request a different source later.

Added a event handler to rotate the light. Rotating the light allows for exactly placement of the light rays.

players can turn off the changes if they disable parameter.
### Please specify which Issue this PR Resolves.
closes #1620 

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

The rotate item function is not upvector safe. 

This pull request is intended to make playing in the night easier for the early game when night vision is unavailable.
An example of the QOL improvement would be looking at the map at night. Most of the item I stand in front of a vehicle with its lights on, but with this purchasable light I can stand anywhere.  

might need to be tested with two people garaging the light.

### CODE REVIEW

**Comment by Bob-Murphy** [comment]
I have no clue about the content of the code but I can say something regarding the order:
- in the templates you have added the light in between vehicleAA and vehicleBoat
- in fn_compatibilityLoadFaction you have it between the repair truck and the statics
- in fn_initVarServer you have it after the mines but before breachingExplosives

Yes, not all the things are properly ordered but if you would put it after the mines/explosives in all cases it would be more orderly and easier to find because consistency.

File: A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		_return = true;
+		PRESSED_E = true;
+		WAIT_TIME = time + 0.01;
+	};
+
+	if(_key isEqualTo DIK_Q && WAIT_TIME < time) then {
+		_return = true;
+		PRESSED_Q = true;
+		WAIT_TIME = time + 0.01;
+	};
```
Comment (HakonRydland) [inline A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf:62]:
```suggestion sqf
	if(_key isEqualTo DIK_Q && (A3A_LightRotate_EHDB # WAIT_TIME ) < time) then {
		_return = true;
		A3A_LightRotate_EHDB set [ Q_PRESSED , true];
		A3A_LightRotate_EHDB set [ WAIT_TIME , (A3A_LightRotate_EHDB # WAIT_TIME ) + 0.01];
	};
```

File: A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+	if(_key in [DIK_SPACE,DIK_RETURN]) then{
+		_return = true;
+		call endRotating;
```
Comment (HakonRydland) [inline A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf:66]:
```suggestion sqf
              call (A3A_LightRotate_EHDB # END_ROTATING );
```

File: A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	};
+	_return;
+}];
+
+FRAME_HANDLER_UTILITY= addMissionEventHandler ["EachFrame", {
```
Comment (HakonRydland) [inline A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf:71]: apply similar changes for this eh too

**Review by HakonRydland — CHANGES_REQUESTED** [review]
some changes needed, mostly fine, as a general note you can make all feedback stringtable entries following the naming scheme given in the comments.
also tabs -> spaces (4 spaces = 1 tab)

File: A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+	//variables
+	PRESSED_E = nil;
+	PRESSED_Q = nil;
+	WAIT_TIME = nil;
+	DIR_OBJECT = nil;
+	LIGHT = nil;
+	INFO_TEXT = nil;
+	CONTROL_HINT = nil;
+};
```
[the remark this answers is not in the document]
Comment (killerswin2) [reply A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf:44]: there is an error here, missing a semicolon. I can't see where there is a missing semi colon.

[the remark this answers is not in the document]
Comment (killerswin2) [reply A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf:44]: locality issue. A3A_light is private. needs underscore.

[the remark this answers is not in the document]
Comment (CalebSerafin) [reply A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf:44]: It's a red-herring error that is thrown when attempting to create a private variable without an underscore. Look at line 32. Usually occurs when changing between global and local variables or from refactorings.

[the remark this answers is not in the document]
Comment (HakonRydland) [reply A3-Antistasi/functions/UtilityItems/fn_rotateItem.sqf:44]: frost is correct i forgot to remove the private as i wrote it as a local variable before checking and realising that display events could be passed additional arguments

**Comment by killerswin2** [comment]
updated all

**Review by HakonRydland — APPROVED** [review]
should be good now

**Pull request merged**